File: typescript-client-sdk/src/sdk/models/operations/getoneconnection.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class GetOneConnectionQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=connectionId" })
  connectionId: string;
}


export class GetOneConnectionResponseBody extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=connection" })
  connection?: shared.Connection;
}


export class GetOneConnectionRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  queryParams: GetOneConnectionQueryParams;
}


export class GetOneConnectionResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  responseBody?: GetOneConnectionResponseBody;
}

File: go-client-sdk/pkg/models/operations/getoneconnection.go
package operations

import (
	"github.com/speakeasy-sdks/orb-sdks/go-client-sdk/pkg/models/shared"
)

type GetOneConnectionQueryParams struct {
	ConnectionID string `queryParam:"style=form,explode=true,name=connectionId"`
}

type GetOneConnectionResponseBody struct {
	Connection *shared.Connection `json:"connection,omitempty"`
}

type GetOneConnectionRequest struct {
	QueryParams GetOneConnectionQueryParams
}

type GetOneConnectionResponse struct {
	ContentType  string
	StatusCode   int64
	ResponseBody *GetOneConnectionResponseBody
}
